internal/resources/app_endpoint.go

```diff
@@ -0,0 +1,661 @@
+
+	// Handle OIDC if present
+	if len(appEndpoint.Oidc) > 0 {
+		plan.Oidc = make([]providerschema.AppEndpointOidc, len(appEndpoint.Oidc))
+		for i, oidc := range appEndpoint.Oidc {
+			plan.Oidc[i] = providerschema.AppEndpointOidc{
+				Issuer:        types.StringValue(oidc.Issuer),
+				ClientId:      types.StringValue(oidc.ClientId),
+				UserPrefix:    types.StringPointerValue(oidc.UserPrefix),
+				DiscoveryUrl:  types.StringPointerValue(oidc.DiscoveryUrl),
+				UsernameClaim: types.StringPointerValue(oidc.UsernameClaim),
+				RolesClaim:    types.StringPointerValue(oidc.RolesClaim),
+				Register:      types.BoolPointerValue(oidc.Register),
+				ProviderId:    types.StringPointerValue(oidc.ProviderId),
+				IsDefault:     types.BoolPointerValue(oidc.IsDefault),
+			}
+		}
+	}
+
+	// Handle require_resync
+	if len(appEndpoint.RequireResync) > 0 {
+		requireResyncMap := make(map[string]attr.Value)
+		for scope, collections := range appEndpoint.RequireResync {
+			items := make([]attr.Value, len(collections))
+			for i, name := range collections {
+				items[i] = types.StringValue(name)
+			}
+			requireResyncMap[scope], diags = types.ListValueFrom(
+				ctx,
+				types.StringType,
+				items,
+			)
+			if diags.HasError() {
+				return nil, fmt.Errorf("error converting require_resync for scope %s: %v", scope, diags.Errors())
+			}
+		}
+		plan.RequireResync, diags = types.MapValueFrom(ctx, types.ListType{ElemType: types.StringType}, requireResyncMap)
+		if diags.HasError() {
+			return nil, fmt.Errorf("error converting require_resync: %s", diags.Errors())
+		}
+	} else {
+		plan.RequireResync = types.MapNull(types.ListType{ElemType: types.StringType})
+	}
+
+	return plan, nil
+}
+
+// Update updates an existing App Endpoint.
+func (a *AppEndpoint) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state providerschema.AppEndpoint
+	diags := req.Plan.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var organizationId = state.OrganizationId.ValueString()
+	var projectId = state.ProjectId.ValueString()
+	var clusterId = state.ClusterId.ValueString()
+	var appServiceId = state.AppServiceId.ValueString()
+	var endpointName = state.Name.ValueString()
+
+	body := schemaToAppEndpointRequest(ctx, state)
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s",
+		a.HostURL, organizationId, projectId, clusterId, appServiceId, endpointName)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusNoContent}
+
+	_, err := a.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		body,
+		a.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating app endpoint",
+			fmt.Sprintf("Could not update app endpoint %s: %s", endpointName, err.Error()),
+		)
+		return
+	}
+
+	// Refresh the state after update
+	refreshedState, err := a.refreshAppEndpoint(ctx, cfg, &state)
```

Comment:
you're passing `cfg` as is to `refreshAppEndpoint` .  that is using `PUT`.  you should use `GET`.  

I would recommend not using `cfg` as an arg.  create a new `cfg` with the correct http method eg

https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/internal/resources/project.go#L350